pkgs/development/python-modules/cairo-lang/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, fetchzip
+, buildPythonPackage
+, setuptools
+, ecdsa
+, fastecdsa
+, sympy
+, frozendict
+, marshmallow
+, marshmallow-enum
+, marshmallow-dataclass
+, marshmallow-oneofschema
+, pipdeptree
+, eth-hash
+, web3
+, aiohttp
+, cachetools
+, mpmath
+, numpy
+, prometheus-client
+, typeguard
+, lark
+, pyyaml
+, pytest-asyncio
+, pytest
+, pytestCheckHook
+, gmp
+}:
+
+buildPythonPackage rec {
+  pname = "cairo-lang";
+  version = "0.9.1";
+
+  src = fetchzip {
+    url = "https://github.com/starkware-libs/cairo-lang/releases/download/v${version}/${pname}-${version}.zip";
+    sha256 = "sha256-i4030QLG6PssfKD5FO4VrZxap19obMZ3Aa77p5MXlNY=";
+  };
+
+  # TODO: remove a substantial part when https://github.com/starkware-libs/cairo-lang/pull/88/files is merged.
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace 'frozendict==1.2' 'frozendict>=1.2' \
+      --replace 'lark-parser' 'lark'
+
+    substituteInPlace starkware/cairo/lang/compiler/parser_transformer.py \
+      --replace 'value, meta' 'meta, value' \
+      --replace 'value: Tuple[CommaSeparatedWithNotes], meta' 'meta, value: Tuple[CommaSeparatedWithNotes]'
+    substituteInPlace starkware/cairo/lang/compiler/parser.py \
+      --replace 'standard' 'basic'
+  '';
+
+  propagatedBuildInputs = [
+    aiohttp
+
+    cachetools
+    setuptools
+
+    ecdsa
+    fastecdsa
+    sympy
+    mpmath
+    numpy
+
+    typeguard
+
+    frozendict
+
+    prometheus-client
+
+    marshmallow
+    marshmallow-enum
+    marshmallow-dataclass
+    marshmallow-oneofschema
+
+    pipdeptree
+
+    lark
+
+    web3
+    eth-hash
+    eth-hash.optional-dependencies.pycryptodome
+    pyyaml
+
+    pytest-asyncio
+    pytest
+  ];
+
+  buildInputs = [ gmp ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  # There seems to be no test included in the ZIP release…
+  doCheck = false;
```

Comment:
Can we clone from github?

Reply:
We can, but it would change the whole build process as it uses CMake upstream with custom CMake modules to produce Python wheels, plus, I'm not sure it is very Nix-friendly (they invoke pip without any concern).

Reply:
@SuperSandro2000 If you are fine with this package, I will revisit in a later version, `cairo-lang` is planning to release a new one in ~4-6 months, apparently, it may be easier to package it from then.

Reply:
I don't know but should be fine for now I hope.